src/main/java/io/neonbee/entity/EntityVerticle.java

```diff
@@ -254,30 +263,127 @@ private Future<Void> announceEntityVerticle(Vertx vertx) {
+
+        String clusterNodeId = HazelcastHelper.getHazelcastNodeId(vertx);
+        LOGGER.debug("Acquire lock for cluster node ID: {}", clusterNodeId);
+        return vertx.sharedData().getLock(clusterNodeId)
+                .onFailure(cause -> LOGGER.error("Failed to acquire lock for cluster node ID: {}", clusterNodeId))
+                .compose(lock -> getNoteEntitiesMap(vertx).compose(map -> map.get(clusterNodeId)
+                        .map(jsonArray -> jsonArray == null ? new JsonArray() : (JsonArray) jsonArray)
+                        .compose(jsonArray -> {
+                            JsonObject jo = new JsonObject().put(QUALIFIED_NAME, qualifiedName).put(ENTITY_NAME, name);
+
+                            if (!jsonArray.contains(jo)) {
+                                jsonArray.add(jo);
+                                return map.put(clusterNodeId, jsonArray)
+                                        .onSuccess(event -> LOGGER.info(
+                                                "registered, cluster node ID: {}, Entity name: {}, qualified name: {} ",
+                                                clusterNodeId, name, qualifiedName));
+                            }
+                            return succeededFuture();
+                        })).onComplete(unused -> {
+                            LOGGER.debug("Release lock for cluster node ID: {}", clusterNodeId);
+                            lock.release();
+                        }));
+    }
+
+    /**
+     * Unregister the entity models for the node with the ID nodeId.
+     *
+     * @param vertx  The Vert.x instance
+     * @param nodeId hazelcast node ID
+     * @return Future
+     */
+    public static Future<Void> unregister(Vertx vertx, String nodeId) {
+        if (!vertx.isClustered()) {
+            return succeededFuture();
+        }
+
+        LOGGER.info("unregistering entity verticle models ...");
+        Future<Void> unregisterFuture = getNoteEntitiesMap(vertx).compose(map -> map.get(nodeId))
+                .map(jsonArray -> jsonArray == null ? new JsonArray() : (JsonArray) jsonArray)
+                .compose(entries -> CompositeFuture.all(removeSharedMapEntries(vertx, entries)).mapEmpty());
+
+        unregisterFuture.onSuccess(unused -> LOGGER.info("unregistered entity verticle models"))
+                .onFailure(cause -> LOGGER.error("failed to unregistered entity verticle models", cause));
+
+        return unregisterFuture;
+    }
+
+    @VisibleForTesting
+    static Future<AsyncMap<Object, Object>> getNoteEntitiesMap(Vertx vertx) {
+        SharedDataAccessor sharedDataAccessor = new SharedDataAccessor(vertx, EntityVerticle.class);
+        return sharedDataAccessor.getAsyncMap("register");
+    }
+
+    private static List<Future> removeSharedMapEntries(Vertx vertx, JsonArray entries) {
+        AsyncMap<String, Object> sharedMap = NeonBee.get().getAsyncMap();
+        List<Future> futureList = new ArrayList<>(entries.size());
```

Comment:
This is not possible because the returned list is used to create a CompositeFuture with CompositeFuture.all. I mean, I could do this, but then I have to cast the list when I call CompositeFuture.all.